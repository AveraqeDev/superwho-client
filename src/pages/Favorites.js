import React from 'react';

import '../styles/Favorites.css';

const FavoritesHeader = () => {
  return (
    <div className="FavoritesHeader">
      <h2>Favorites</h2>
    </div>
  )
};

const TableControls = () => {
  return (
    <div className="TableControls">
      <p>Showing 10 of 113</p>
      <div className="PageButtons">
        <button className="TableControls__button prev">
          {'< Prev'}
        </button>
        <button className="TableControls__button next">
          {'Next >'}
        </button>
      </div>
    </div>
  );
};

const Favorites = () => {
  return (
    <div className="Favorites">
      <FavoritesHeader />
      <TableControls />
    </div>
  );
};

export default Favorites;