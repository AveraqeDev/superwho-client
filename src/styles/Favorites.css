.Favorites {
  margin-left: 15%;
  padding: .5rem;
}


@media screen and (max-width: 900px) {
  .Favorites {
    margin-left: 20%;
  }
}

@media screen and (max-width: 760px) {
  .Favorites {
    margin-left: 0;
  }
}