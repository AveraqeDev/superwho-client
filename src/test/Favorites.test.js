import React from 'react';
import ReactDOM from 'react-dom';
import renderer from 'react-test-renderer';
import Favorites, { FavoritesHeader } from '../pages/Favorites';

describe('Favorites', () => {
  describe('<FavoritesHeader />', () => {
    // Smoke test
    it('renders without crashing', () => {
      const div = document.createElement('div');
      ReactDOM.render(<FavoritesHeader />, div);
      ReactDOM.unmountComponentAtNode(div);
    });

    //Snapshot test
    it('renders the UI as expected', () => {
      const tree = renderer
        .create(
          <FavoritesHeader />
        )
        .toJSON();
      expect(tree).toMatchSnapshot();
    })
  });

  describe('<Favorites />', () => {
    // Smoke test
    it('renders without crashing', () => {
      const div = document.createElement('div');
      ReactDOM.render(<Favorites />, div);
      ReactDOM.unmountComponentAtNode(div);
    });

    //Snapshot test
    it('renders the UI as expected', () => {
      const tree = renderer
        .create(
          <Favorites />
        )
        .toJSON();
      expect(tree).toMatchSnapshot();
    })
  });
});