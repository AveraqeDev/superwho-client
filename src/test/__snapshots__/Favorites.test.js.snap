// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Favorites <Favorites /> renders the UI as expected 1`] = `
<div
  className="Favorites"
>
  <div
    className="FavoritesHeader"
  >
    <h2>
      Favorites
    </h2>
  </div>
  <h2 />
</div>
`;

exports[`Favorites <FavoritesHeader /> renders the UI as expected 1`] = `
<div
  className="FavoritesHeader"
>
  <h2>
    Favorites
  </h2>
</div>
`;
